fri/src/verifier.rs

```diff
@@ -39,48 +49,173 @@ pub struct FriVerifier<
 impl<F: FftField + PrimeField, P: Prng + Clone + 'static, W: Digest + Clone + 'static>
     FriVerifier<F, P, W>
 {
-    pub fn new(
+    pub fn new<C>(
         channel: FSVerifierChannel<F, P, W>,
         params: FriParameters<F, Radix2EvaluationDomain<F>>,
         commitment_hashes: CommitmentHashes,
-        first_layer_callback: FirstLayerQueriesCallback<F>,
+        first_layer_callback: C,
+        query_indices_test: Option<Vec<u64>>,
+        to_verify_test: Option<Vec<BTreeMap<RowCol, F>>>,
         eval_points_test: Option<Vec<F>>,
         last_layer_coefficients_test: Option<Vec<F>>,
-    ) -> Self {
+    ) -> Self
+    where
+        C: Fn(&[u64]) -> Vec<F> + 'static,
+    {
         let n_layers = params.fri_step_list.len();
         Self {
             channel,
             params,
             commitment_hashes,
-            first_layer_callback,
+            first_layer_callback: Box::new(first_layer_callback),
             n_layers,
             first_eval_point: None,
             eval_points: vec![F::zero(); n_layers - 1],
```

Comment:
Can this just be an empty Vec to avoid "sentinel" values: better to crash with out-of-bounds that accidentally use a zero challenge without knowing it